agentMET4FOF_tutorials/tutorial_4_metrological_agents_and_streams.py

```diff
@@ -0,0 +1,87 @@
+import time
+from typing import Dict
+
+import numpy as np
+from time_series_metadata.scheme import MetaData
+
+from agentMET4FOF.agents import AgentNetwork
+from agentMET4FOF.metrological_agents import MetrologicalAgent, MetrologicalMonitorAgent
+from agentMET4FOF.metrological_streams import MetrologicalSineGenerator
+
+
+class MetrologicalSineGeneratorAgent(MetrologicalAgent):
+    """An agent streaming a sine signal
+
+    Takes samples from the :py:mod:`SineGenerator` and pushes them sample by sample
+    to connected agents via its output channel.
+    """
+
+    # The datatype of the stream will be SineGenerator.
+    _sine_stream: MetrologicalSineGenerator
+
+    def init_parameters(self, signal: MetrologicalSineGenerator = MetrologicalSineGenerator(), **kwargs):
+        """Initialize the input data
+
+        Initialize the input data stream as an instance of the
+        :py:mod:`SineGenerator` class
+
+        Parameters
+        ----------
+        signal : Signal
```

Comment:
The class `Signal` does not exist here, it is only provided in some of the tests. Either this should be `MetrologicalSineGenerator` or its base class `MetrologicalDataStreamMET4FOF`. Since we are using the metadata property we should not request a `DataStreamMET4FOF` object though.

Reply:
True. It should be MetrologicalSineGenerator.